esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1447,7 +1447,21 @@ void EsdlServiceImpl::prepareFinalRequest(IEspContext &context,
     if (serviceCrt || methodCrt)
     {
         context.addTraceSummaryTimeStamp(LogNormal, "srt-custreqtrans");
-        processServiceAndMethodTransforms({serviceCrt, methodCrt}, &context, tgtcfg, srvdef, mthdef, reqProcessed, m_oEspBindingCfg.get());
+
+        //as we add more entry points the script context will move to wider scope
+        Owned<IEsdlScriptContext> scriptContext = createEsdlScriptContext(&context);
+        scriptContext->setContent("ESDLRequest", reqProcessed.str());
+        scriptContext->setAttribute("esdl", "service", srvdef.queryName());
+        scriptContext->setAttribute("esdl", "method", mthdef.queryMethodName());
+        scriptContext->setAttribute("esdl", "request_type", mthdef.queryRequestType());
+        scriptContext->setAttribute("esdl", "request", mthdef.queryRequestType());
```

Comment:
Why do we have different attributes for request and request_type?

Reply:
We use them as the same right now but I could see them sometimes changing.  We just happen to support the request tag that is named after the typename.  May never need to separate but I wanted to make it clear that they may not always be the same.